tests/test-awful-client.lua

```diff
@@ -79,4 +77,87 @@ end
 end
 }
 
+local has_error
+
+table.insert(steps, function()
+    -- Make sure there is no extra callbacks that causes double screen changes
+    -- (regress #1028 #1052)
+    client.connect_signal("property::screen",function(c)
+        if c.class and c.class ~= "screen"..c.screen.index then
+            has_error = table.concat {
+                "An unexpected screen change did occur\n",
+                debug.traceback(),"\n",
+                c.class, " Screen: ",c.screen.index
+            }
```

Comment:
Because it will happen in normal test execution. It is only an error if it happens during certain steps. I know in *this* PR, it isn't the case anymore. I have more tests, and then it became a problem and its why I did it. However, the final version of this PR doesn't include them (well, I moved one of them to the tag tests and the other isn't ready**). I could just disconnect the signal. It would work too.

\** test titlebar drag and drop (move or resize) across screens with tags selected, no tags selected and no tags at all.